media/filters/chunk_demuxer.h

```diff
@@ -39,9 +42,15 @@ class MEDIA_EXPORT ChunkDemuxerStream : public DemuxerStream {
  public:
   using BufferQueue = base::circular_deque<scoped_refptr<StreamParserBuffer>>;
 
+#if BUILDFLAG(USE_STARBOARD_MEDIA)
```

Comment:
This style usually actually ends up harder to maintain and re-merge. I'd advise against this.

Reply:
Then I'd suggest just adding |mime_type| as the third parameter and give it a default value. Having two fully separate constructors that look so much alike is also strange to me and might end up needing maintenance, even if rebasing is easy (rebase might work automatically and end up not working because the original ctor has deviated).

@osagie98 could all this be upstreamed BTW?

Reply:
We have a general approach of `#ifdef` -gating ( or hopefully at some point, runtime-flag gating ) all changes to upstream code. Yes, adding an arg with a default is cleaner to read, but the change should still be guarded, which gets you back to this place here.

I also support sending this change upstream if applicable, but we still have to resolve this in `m114` for now.